trieste/acquisition/sampler.py

```diff
@@ -412,7 +412,7 @@ def __init__(
 
         try:
             self._noise_variance = model.get_observation_noise()
-            self._kernel = model.get_kernel()
+            self._kernel = model.get_kernel()  # type: ignore
```

Comment:
Might be better to restrict the models the sampler accepts to GPflowPredictor models instead, as other models with a get_kernel method aren't necessarily going to return a gpflow.kernels.Kernel?

Reply:
The issue here is that our dummy test model is not of GPFlowPredictor type, so the type checking gets unhappy in the unit tests.